### PR TITLE
Add horizontal Cartesian components for point masses

### PR DESCRIPTION
**Pull request by birocoles**
Implement horizontal Cartesian components `g_northing`and `g_easting` for point masses. This PR addresses some of #85 . Components in spherical coordinate system will be implemented in the future. 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [x] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

### CODE REVIEW

**Comment by leouieda** [comment]
@birocoles notice that your branch is out of date with respect to the master branch. You might want to merge in the changes before proceeding (you can do this locally or through Github)

File: harmonica/forward/point_mass.py [hunk 1]
```diff
@@ -172,22 +187,16 @@ def point_mass_gravity(
     cast = np.broadcast(*coordinates[:3])
     result = np.zeros(cast.size, dtype=dtype)
     # Prepare arrays to be passed to the jitted functions
-    coordinates = tuple(np.atleast_1d(i).ravel() for i in coordinates[:3])
-    points = tuple(np.atleast_1d(i).ravel() for i in points[:3])
-    masses = np.atleast_1d(masses).ravel()
-    # Sanity checks
```
Comment (birocoles) [inline harmonica/forward/point_mass.py]: This change makes the test `test_invalid_masses_array` useless. I will remove this test.

Comment (santisoler) [reply harmonica/forward/point_mass.py]: Hi @birocoles, sorry for my disappearance, I was a little bit busy on the last weeks.

Why would you remove the sanity check for the point masses?

**Comment by birocoles** [comment]
Hi @santisoler! I may have done something wrong when I tried to pull things from master. Actually, I think the sanity checks are important. Whatever. I will bring this sanity check and the related test back to the code.

**Comment by birocoles** [comment]
@santisoler , I think everything is working now.

**Comment by birocoles** [comment]
I have just fixed some problems with too long lines. What I have to do now?

**Comment by santisoler** [comment]
Thanks for tackling this @birocoles! :1st_place_medal: 

I think this is good to be merged. If you agree, please remove the WIP from the PR title and ask @leouieda for a review so we can merge it.
Thanks again for the hard work!

**Comment by birocoles** [comment]
Thanks @santisoler

**Pull request merged**